### PR TITLE
added new people landing docs.

### PR DESCRIPTION
**Pull request by agladstein**
closes https://github.com/popgensims/stdpopsim/issues/141

### CODE REVIEW

File: docs/development.rst [hunk 1]
```diff
@@ -4,10 +4,42 @@
 Development
 ===========
 
-If you would like to help with ``stdpopsim`` development, please read the
-following. If you think there is anything missing,
-please open an `issue <http://github.com/popgensims/stdpopsim/issues>`_ or
-`pull request <http://github.com/popgensims/stdpopsim/pulls>`_ on GitHub!
+``stdpopsim`` **is a community effort, and YOU are welcome to join in!**
+
+We envision two main types of ``stdpopsim`` developers:
```
Comment (andrewkern) [inline docs/development.rst]: Maybe change to "at least two main types"?

File: docs/development.rst [hunk 2]
```diff
@@ -4,10 +4,42 @@
...
+
+1. Model contributors
+2. API developers
+
+`Model contributors` add simulation code of published models. This could be your
+own published model or any other published model you think would be useful.
+This is the main way we expect biologists to continually add to the catalog
+of models. See the section `Model review process`_.
```
Comment (andrewkern) [inline docs/development.rst]: worth adding what API developers might do? maybe mention that we are looking for folks to help with documentation / tutorials too?

Comment (agladstein) [reply docs/development.rst]: Yeah, I wasn't really sure what to put for that. I just have "API developers work on the internal code base". Do you have a suggestion of what else I should add about API developers?
Are you thinking of helping with documentation/tutorials in the same category as API developers or different category?

Comment (andrewkern) [reply docs/development.rst]: i was thinking a separate category? don't really have a strong opinion but i think we want to shrink the barriers to entry as much as possible.

As for API devs hows this:

 "API developers work on infrastructure development for the PopSim Consortium, which could include improvements and additions to the internal code base of `stdpopsim`, establishment of benchmarking pipelines, and new projects that align with consortium goals."

Comment (agladstein) [reply docs/development.rst]: Just pushed some edits. See what you think.

Comment (andrewkern) [reply docs/development.rst]: lgtm. let's merge!

**Pull request merged**